addons/l10n_de/models/datev.py

```diff
@@ -1,12 +1,45 @@
-from odoo import fields, models
-
+from odoo import fields, models, api, _
+from odoo.exceptions import UserError
 
 class AccountTax(models.Model):
     _inherit = "account.tax"
 
     l10n_de_datev_code = fields.Char(size=4, help="4 digits code use by Datev")
 
 
+class AccountMove(models.Model):
+    _inherit = 'account.move'
+
+    partner_country_code = fields.Char(related='partner_id.country_code', readonly=True)
+    delivery_date = fields.Date(
+        string='Delivery Date',
+        copy=False,
+        store=True,
+        compute='_compute_delivery_date',
+    )
+
+    @api.depends('invoice_date')
+    def _compute_delivery_date(self):
+        for move in self:
+            if move.invoice_date and not move.delivery_date:
+                move.delivery_date = move.invoice_date
+
+    def _post(self, soft=True):
+        # OVERRIDE to check the invoice lines taxes.
+        for invoice in self.filtered(lambda move: move.is_invoice()):
+            if (self.partner_country_code == 'DE' or self.env.company.account_fiscal_country_id.code == 'DE') and (not invoice.delivery_date):
+                raise UserError(_('You must set a delivery date on the invoice for Germany.'))
+            for line in invoice.invoice_line_ids:
+                account_tax = line.account_id.tax_ids.ids
+                if account_tax and invoice.company_id.account_fiscal_country_id.code == 'DE':
+                    account_name = line.account_id.name
+                    for tax in line.tax_ids:
+                        if tax.id not in account_tax:
+                            raise UserError(_('Account %s does not authorize to have tax %s specified on the line. \
+                                Change the tax used in this invoice or remove all taxes from the account') % (account_name, tax.name))
```

Comment:
What is this ? Why ???

Reply:
I actually just wanted this, must've done something when rebasing or there's a rebase missing, will fix it.
```python
for invoice in self.filtered(lambda move: move.is_invoice()):
            if (self.partner_country_code == 'DE' or self.env.company.account_fiscal_country_id.code == 'DE') and (not invoice.delivery_date):
                raise UserError(_('You must set a delivery date on the invoice for Germany.'))
```

Reply:
Because that part is older code no longer used